Ginger/GingerCoreCommon/VariablesLib/VariableNumber.cs

```diff
@@ -0,0 +1,242 @@
+        {
+            get { return true; }
+        }
+
+        public override string VariableUIType
+        {
+            get { return GingerDicser.GetTermResValue(eTermResKey.Variable) + " Number"; }
+        }
+
+        [IsSerializedForLocalRepository]
+        public string InitialNumberValue
+        {
+            set
+            {
+                if(!mIsDecimalValue)
+                {
+                    mInitialNumberValue = GetValidInteger(value).ToString();
+                }
+                else
+                {
+                    mInitialNumberValue = GetFloatWithPrecisionValue(value);
+                }
+
+                Value = value.ToString();
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+            get
+            {
+                if(mInitialNumberValue == null)
+                {
+                    mInitialNumberValue = "0";
+                }
+                return mInitialNumberValue;
+            }
+        }
+
+        private string GetFloatWithPrecisionValue(string value)
+        {
+            if (value.Contains(".") && value.Split('.')[1].Length > 0)
+            {
+                float validFloat;
+                float.TryParse(value, out validFloat);
+                if (validFloat < mMinValue || validFloat > mMaxValue)
+                {
+                    Reporter.ToLog(eLogLevel.WARN, "Input value is out of range!");
+                    return mInitialNumberValue;
+                }
+                return Math.Round(Convert.ToDouble(validFloat), Convert.ToInt32(mPrecisionValue)).ToString();
+            }
+            else
+            {
+                return value;
+            }
+        }
+
+        private  string GetValidInteger(string value)
+        {
+            try
+            {
+                float validInteger;
+                float.TryParse(value, out validInteger);
+                if (validInteger < mMinValue || validInteger > mMaxValue)
+                {
+                    Reporter.ToLog(eLogLevel.WARN, "Input value is out of range!");
+                    return mInitialNumberValue;
+                }
+                return Convert.ToInt32(validInteger).ToString();
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error occured during GetValidInteger..", ex);
+                return mInitialNumberValue;
+            }
+
+        }
+
+        [IsSerializedForLocalRepository]
+        public int? PrecisionValue
+        {
+            get
+            {
+                return mPrecisionValue;
+            }
+            set
+            {
+                mPrecisionValue = value;
+                OnPropertyChanged("PrecisionValue");
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+        }
+
+        public override void GenerateAutoValue()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5355677)